drivers/imx-enet-regs.h

```diff
@@ -305,7 +305,7 @@ struct enet_regs {
 typedef struct
 {
 	uint16_t len; /* for last frag = whole frame size? (TBV) */
-	uint16_t flags;
+	_Atomic volatile uint16_t flags;
```

Comment:
- this is read/written by the hardware, so changing this to atomic doesn't do the expected thing
- `volatile` is also not needed, as we're always declare the full struct as `volatile`

Reply:
- it's also read and written by the driver using `atomic_store`, so shouldn't it be atomic?

Reply:
have you checked (`git blame`) WHY it's being written using `atomic_store`? IMHO it needs additional comment (regarding WHY we're using `atomic_store` in certain places) instead of type change.

Not sure if `atomic_store` is really needed there, but I assume that it came up in testing.

Reply:
ok, I see it's for memory synchronization, will restore the type changed and add an appropriate comment

Reply:
exactly :+1: